src/vmm/src/vmm_config/mod.rs

```diff
@@ -177,6 +183,258 @@ fn open_file_nonblock(path: &Path) -> Result<File, std::io::Error> {
         .open(path)
 }
 
+// TODO: See below doc comment.
+/// Mimic of `log::Level`.
+///
+/// This is used instead of `log::Level` to support:
+/// 1. Aliasing `Warn` as `Warning` to avoid a breaking change in the API (which previously only
+///    accepted `Warning`).
+/// 2. Setting the default to `Warn` to avoid a breaking change.
+///
+/// This alias, custom `Default` and type should be removed in the next breaking update to simplify
+/// the code and API (and `log::Level` should be used in place).
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub enum Level {
+    /// The “error” level.
+    ///
+    /// Designates very serious errors.
+    #[serde(alias = "ERROR")]
+    Error,
+    /// The “warn” level.
+    ///
+    /// Designates hazardous situations.
+    #[serde(alias = "WARNING", alias = "Warning")]
+    Warn,
+    /// The “info” level.
+    ///
+    /// Designates useful information.
+    #[serde(alias = "INFO")]
+    Info,
+    /// The “debug” level.
+    ///
+    /// Designates lower priority information.
+    #[serde(alias = "DEBUG")]
+    Debug,
+    /// The “trace” level.
+    ///
+    /// Designates very low priority, often extremely verbose, information.
+    #[serde(alias = "TRACE")]
+    Trace,
+}
+impl Default for Level {
+    fn default() -> Self {
+        Self::Warn
+    }
+}
+impl From<Level> for tracing::Level {
+    fn from(level: Level) -> tracing::Level {
+        match level {
+            Level::Error => tracing::Level::ERROR,
+            Level::Warn => tracing::Level::WARN,
+            Level::Info => tracing::Level::INFO,
+            Level::Debug => tracing::Level::DEBUG,
+            Level::Trace => tracing::Level::TRACE,
+        }
+    }
+}
+impl From<log::Level> for Level {
+    fn from(level: log::Level) -> Level {
+        match level {
+            log::Level::Error => Level::Error,
+            log::Level::Warn => Level::Warn,
+            log::Level::Info => Level::Info,
+            log::Level::Debug => Level::Debug,
+            log::Level::Trace => Level::Trace,
+        }
+    }
+}
+impl FromStr for Level {
+    type Err = <log::Level as FromStr>::Err;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        // This is required to avoid a breaking change.
+        match s {
+            "ERROR" => Ok(Level::Error),
+            "WARNING" | "Warning" => Ok(Level::Warn),
+            "INFO" => Ok(Level::Info),
+            "DEBUG" => Ok(Level::Debug),
+            "TRACE" => Ok(Level::Trace),
+            _ => log::Level::from_str(s).map(Level::from),
+        }
+    }
+}
+
+/// Strongly typed structure used to describe the logger.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct LoggerConfig {
+    /// Named pipe or file used as output for logs.
+    pub log_path: std::path::PathBuf,
+    /// The level of the Logger.
+    pub level: Option<Level>,
```

Comment:
All API calls that previously work will still work, we are not breaking the API. This is more of an implementation detail. Rather than having serde insert a default value if one isn't found when deserializing, we allow it to be `None` then we insert the default value with `.unwrap_or(MY_DEFAULT)`.